sdks/go/pkg/beam/core/runtime/exec/hash.go

```diff
@@ -101,13 +109,18 @@ func (h *stringHasher) Hash(element interface{}) (uint64, error) {
 	n := l - i
 	copy(b[:], s[i:])
 	h.hash.Write(b[:n])
+	h.we.EncodeSingle(w, h.hash)
 	return h.hash.Sum64(), nil
 }
 
 type numberHasher struct {
+	hash  hash.Hash64
+	we    WindowEncoder
+	cache []byte
```

Comment:
Nit: Cache seems to need a fixed size of 8 bytes for the Hash method to work correctly, so I'd just initialize it in a newNumberHasher() function or something similar, instead of leaving it to calling code.

Reply:
Good call.